api/CatFacts.go

```diff
@@ -0,0 +1,48 @@
+package api
+
+import (
+	"net/http"
+	"errors"
+	"strings"
+)
+
+const catFactsBaseUrl = "https://catfact.ninja"
+
+const catFactEndpoint = catFactsBaseUrl + "/fact"
+const catFactFactKey = "fact"
+
+func GetCatFact() (string, error) {
+	resp, err := http.Get(catFactEndpoint)
+
+	if (err != nil) {
+		return "", err
+	}
+
+	defer resp.Body.Close()

```

Comment:
something I started seeing at work is people putting typical `defer`ed functions in a wrapper that catches errors for them. Since `Close()` can return an error yet its typically `defer`ed it causes room for missed bugs. So what i've seen some people do is

```
func logError(f func () error) {
	err := f()
	if err != nil {
		log15.Error("something happened!", "error", err)
	}
}
```

then instead of `defer resp.Body.Close()` they do `defer logError(resp.Body.Close)`

Reply:
Just want your opinion on it. I think it does nothing but help, even if you don't have all the context you normally would have when logging errors